book-service-rest/src/main/java/com/example/Library/controller/UsersController.java

```diff
@@ -0,0 +1,106 @@
+    @Autowired
+    private LoginAttemptService loginAttemptService;
+
+    @Autowired
+    public UsersController(UserService userService, UserRepository userRepository) {
+        this.userService = userService;
+        this.userRepository = userRepository;
+    }
+   // @PreAuthorize("hasAuthority('books:create')")
+    @GetMapping()
+    @Operation(summary = "Get all blocked users")
+    public ArrayList<String> getAllBlockedUsers(){
+        return loginAttemptService.getAllBlockedUsers();
+    }
+
+    @PreAuthorize("hasAuthority('books:create')")
+    @PostMapping()
+    @Operation(summary = "Add a user to the library using parameters.")
+    public UserDTO addUser(@RequestParam(value = "firstName") String firstName,
+                           @RequestParam(value = "lastName") String lastName,
+                           @RequestParam(value = "age") int age,
+                           @RequestParam(value = "email") String email,
+                           @RequestParam(value = "password") String password,
+                           @RequestParam(value = "gender") String gender,
+                           @RequestParam(value = "role") String role,
+                           @RequestParam(value = "userStatus") String userStatus) {
+        log.info("Executing method POST /api/v1/users with params : firstName = " + firstName + ",lastName = " + lastName + ",age = " + age + ",email = " + email
+                + ",password = " + password + ",gender = " + gender + ",role = " + role + ",userStatus = " + userStatus);
+        boolean checkUser = userRepository.existsByEmail(email);
+        if (checkUser) {
+            log.error("User with email " + email + " already exists");
+            throw new UserAlreadyExists(email);
+        }
+        if (!gender.equals("MALE") && !gender.equals("FEMALE")) {
+            log.error("You set incorrect user gender : " + gender + ". At this moment we support 2 genders: MALE and FEMALE. Please choose one of these.");
+            throw new IncorrectGenderException(gender);
+        }
+        if (!role.equals("ADMIN") && !role.equals("CLIENT") && !role.equals("GUEST")) {
+            log.error("You set incorrect user role : " + role + ". At this moment we support 3 roles: ADMIN, CLIENT and GUEST. Please choose one of these.");
+            throw new IncorrectRoleException(role);
+        }
+        if (!userStatus.equals("ACTIVE") && !userStatus.equals("BANNED")) {
+            log.error("You set incorrect user status : " + userStatus + ". At this moment we support 2 statuses: ACTIVE and BANNED. Please choose one of these.");
+            throw new IncorrectUserStatusException(userStatus);
+        }
+
+        UserDTO userDTO = new UserDTO();
+        userDTO.setFirstName(firstName);
+        userDTO.setLastName(lastName);
+        userDTO.setAge(age);
+        userDTO.setEmail(email);
+        userDTO.setPassword(passwordEncoder.encode(password));
+        if (gender.equals("MALE")) {
```

Comment:
It might be easier to have an enum or Set<String> with allowed values, so you can check if the input value is allowed, and than set if allowed, otherwise throw validation exception